Source/WebCore/css/ShorthandSerializer.cpp

```diff
@@ -1315,6 +1319,35 @@ String ShorthandSerializer::serializeTextWrap() const
     return makeString(nameLiteral(mode), ' ', nameLiteral(style));
 }
 
+String ShorthandSerializer::serializeSingleAnimationRange(CSSValue& value, TimelineRangeStartOrEnd startOrEnd, CSSValueID startValueID) const
+{
+    if (auto* pair = dynamicDowncast<CSSValuePair>(value)) {
+        if (SingleTimelineRange::isDefault(downcast<CSSPrimitiveValue>(pair->second()), TimelineRangeStartOrEnd::Start) && (startOrEnd == TimelineRangeStartOrEnd::Start || pair->first().valueID() != startValueID))
```

Comment:
I think it would be worth splitting this into variables like for ComputedStyleExtractor, or at least adding some comments describing the different cases.

Reply:
Fixed.